konf.py

```diff
@@ -63,11 +63,29 @@ def is_apex_domain(context, s):
 
 
 class Konf:
-    def __init__(self, values_file, env, local_qa, docker_tag, overrides):
+    def __init__(
+        self, values_file, env, local_qa, docker_tag, database_url, overrides
+    ):
         self.deployment_env = env
+        self.values_file = values_file
+        self.local_qa = local_qa
+        self.docker_tag = docker_tag
+        self.database_url = database_url
+        self.overrides = overrides
 
         # Load project data
-        self.load_values(values_file, local_qa, docker_tag, overrides)
+        self.load_values()
+
+        # Try to replace database URL
+        if database_url and (environemt := self.values.get("env")):
```

Comment:
This is the first walrus I've seen in the wild

Reply:
could this be plural? `environments` or `envs`? And then in the for loop below you could use the singular form of the same - so `for env in envs:`

Reply:
It can, I'll update in a sec.

Reply:
I've been thinking about this and I'm personally not a fan of this way of using the Walrus operator (`:=`). I think this would always be much clearer for people to understand as:

``` python3
environment = self.values.get("env")
if database_url and environment:
```

The example given in [PEP 572](https://www.python.org/dev/peps/pep-0572/) for the walrus operator is instead in a while loop:

``` python3
while chunk := file.read(8192):
   process(chunk)
```

Here I think it genuinely makes things clearer, but I don't think it does so in `if` statements, I think it makes things more obscure.

But it's up to you, I won't block the PR on this point.